src/Module.ts

```diff
@@ -308,20 +308,28 @@ export default class Module {
 		);
 	}
 
-	getReexports() {
+	getReexports(walkedModuleIds = new Set<string>()) {
 		const reexports = Object.create(null);
 
 		for (const name in this.reexports) {
 			reexports[name] = true;
 		}
 
+		// avoid infinite recursion when using circular `export * from X`
+		if (walkedModuleIds.has(this.id)) {
+			return Object.keys(reexports);
```

Comment:
Instead of the `Object.keys`, couldn't you just return `this.reexports`? Then you could move the bail-out condition to the first line of the method. I even think that returning an empty array instead of `this.reexports` should work as well as the reexports of this module are already handled.

Reply:
Good catch! I updated the commit.